core/drivers/pl022_spi.c

```diff
@@ -184,6 +194,14 @@ static void pl022_txrx8(struct spi_chip *chip, uint8_t *wdat, uint8_t *rdat,
 			if (read8(pd->base + SSPSR) & SSPSR_RNE)
 				/* rx 1 packet */
 				rdat[j++] = read8(pd->base + SSPDR);
+			else
+				cnt++;
+
+			if (cnt > SSP_LOOP_TIMEOUT) {
```

Comment:
How can this be reliable on all platforms? (there is no timer or time function involved). Same for the 16-bit version below.